### PR TITLE
Split `utils/yaml.py::read_yaml()`...

### PR DESCRIPTION
**Pull request by ben-alkov**
...into `read_yaml()` and `validate_yaml()`, making it possible to validate without reading
(for e.g. YAML not read from a file)

Signed-off-by: Ben Alkov <ben.alkov@redhat.com>

CLOUDBLD-2830

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/utils/yaml.py [hunk 1]
```diff
@@ -50,12 +53,22 @@ def read_yaml(yaml_data, schema, package=None):
...
     except ValueError:
         logger.error('unable to decode JSON schema, cannot validate')
         raise
-    data = yaml.safe_load(yaml_data)
+
+    validate_with_schema(data, schema)
+    return data
+
+
+def validate_with_schema(data, schema):
```
Comment (MartinBasti) [inline osbs/utils/yaml.py:67]: Should have unittest, it's a new public method

File: osbs/utils/yaml.py [hunk 2]
```diff
@@ -50,12 +53,22 @@ def read_yaml(yaml_data, schema, package=None):
         logger.error('unable to extract JSON schema, cannot validate')
         raise
 
+    # Load schema into py object
     try:
         schema = json.load(schema)
     except ValueError:
         logger.error('unable to decode JSON schema, cannot validate')
         raise
-    data = yaml.safe_load(yaml_data)
```
Comment (MartinBasti) [inline osbs/utils/yaml.py:58]: why did you changed order (schema check, data loading)?

File: osbs/utils/yaml.py [hunk 3]
```diff
@@ -50,12 +53,22 @@ def read_yaml(yaml_data, schema, package=None):
...
+
+    validate_with_schema(data, schema)
+    return data
+
+
+def validate_with_schema(data, schema):
+    """
+    :param data: object, data to be validated
```
Comment (tkdchen) [inline osbs/utils/yaml.py:69]: Type of `data` could be more specific here.

**Review by tkdchen — COMMENTED** [review]
Hi @ben-alkov , I'm a little confused. To validate a YAML file with a JSON schema, before calling the jsonschema's `validate`  method, the YAML file has to be loaded into a `dict` object. So, why is `making it possible to validate without reading` possible?

File: osbs/utils/yaml.py (hunk 2 above)
Comment (ben-alkov) [reply osbs/utils/yaml.py:58]: I reasoned that there's no point in messing with the schema at all if there's an issue with loading the data to be verified.

File: osbs/utils/yaml.py (hunk 3 above)
Comment (ben-alkov) [reply osbs/utils/yaml.py:69]: It's whatever `yaml.safe_load()` returns; IIUC, it's a Python Dict. I'll modify the comment.

**Comment by ben-alkov** [comment]
> Hi @ben-alkov , I'm a little confused. To validate a YAML file with a JSON schema, before calling the jsonschema's `validate` method, the YAML file has to be loaded into a `dict` object. So, why is `making it possible to validate without reading` possible?

Apologies, that should be SCHEMA not read from a file.

**Review by tkdchen — APPROVED** [review]

File: osbs/utils/yaml.py [hunk 4]
```diff
@@ -50,12 +62,20 @@ def read_yaml(yaml_data, schema, package=None):
         logger.error('unable to extract JSON schema, cannot validate')
         raise
 
+    # Load schema into Dict
     try:
         schema = json.load(schema)
-    except ValueError:
+    except (ValueError):
```
Comment (MartinBasti) [inline osbs/utils/yaml.py:68]: :eyes:

**Review by MartinBasti — APPROVED** [review]

File: osbs/utils/yaml.py (hunk 4 above)
Comment (ben-alkov) [reply osbs/utils/yaml.py:68]: Rats

**Pull request merged**